src/cargo/core/resolver/features.rs

```diff
@@ -319,8 +319,32 @@ impl ResolvedFeatures {
         pkg_id: PackageId,
         features_for: FeaturesFor,
     ) -> Vec<InternedString> {
-        self.activated_features_int(pkg_id, features_for)
-            .expect("activated_features for invalid package")
+        let fk = features_for.apply_opts(&self.opts);
+        let key = (pkg_id, fk);
+        if let Some(fs) = self.activated_features.get(&key) {
+            fs.iter().cloned().collect()
+        } else {
+            panic!(
+                "did not find features for {key:?} within activated_features:\n{:#?}",
+                self.activated_features
```

Comment:
I could also print `self.activated_features.keys()` to clean up the output a tiny bit
```diff
...
    (
        PackageId {
            name: "foo",
            version: "0.0.0",
            source: "[ROOT]/foo",
        },
        NormalOrDev,
+    ),
-    ): {},
...
```

Reply:
That sounds great!!